packages/components/src/shared/artplayer.ts

```diff
@@ -0,0 +1,72 @@
+import type artplayer from "artplayer/dist/artplayer.js";
+
+export type ArtPlayer = artplayer;
+export type ArtPlayerOptions = typeof artplayer.option;
+
+type ArtPlayerCustomType = typeof artplayer.option.customType;
+
+export interface CustomArtPlayerOptions
+  extends Omit<ArtPlayerOptions, "container" | "url" | "customType"> {
+  container?: HTMLElement | string;
+  url?: string;
+  quality?: {
+    /**
+     * Whether the default is selected
+     */
+    default?: boolean;
+
+    /**
+     * Html string of quality
+     */
+    html: string;
+
+    /**
+     * Video quality url
+     */
+    url: string;
+    type?: string;
+  }[];
+  customType?:
+    | Record<string, ArtPlayerCustomType>
+    | {
+        [key: string]: ArtPlayerCustomType;
+      }
+    | any;
```

Comment:
Try not use complicate type like this, extract them to interface and export them as users may need to use them.